hyperjumptech/monika
# Feat: Verify config file with json schema

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing #878 

## How did you implement / how did you fix it  
1.  Add validation against json schema
2. Update test configs to comply the schema

## How to test  
1. Create a config with some typo `monika.yml` like this:
```yaml
probes:
  - id: http-probe-01
    name: 'test status 400'
    interval: 5
    requests:
      - url-bad: httpbin.org/status/400
        method: GET
        timeout: 5000
        saveBody: false
    alerts:
      - query: response.time > 10000
        message: Yo too slow!
    incidentThreshold: 2
    recoveryThreshold: 2

notifications:
  - id: unique-id-desktop
    type: desktop
```
3. run the monika with it `bin/run-dev -c monika.yml`
4. You should get an error `CLIError: Probe request URL does not exist`
5. Modify the config for various errors.
6. `npm run test` should all pass

## Screenshot

![image](https://user-images.githubusercontent.com/964106/189799621-51077953-3900-450a-9e0b-297428b4d1e4.png)
